src/networkd.c

```diff
@@ -1122,33 +1122,46 @@ append_wpa_auth_conf(GString* s, const NetplanAuthenticationSettings* auth, cons
     if (auth->anonymous_identity) {
         g_string_append_printf(s, "  anonymous_identity=\"%s\"\n", auth->anonymous_identity);
     }
-    if (auth->password) {
-        if (auth->key_management == NETPLAN_AUTH_KEY_MANAGEMENT_WPA_PSK ||
-            auth->key_management == NETPLAN_AUTH_KEY_MANAGEMENT_WPA_SAE) {
-            size_t len = strlen(auth->password);
-            if (len == 64) {
-                /* must be a hex-digit key representation */
-                for (unsigned i = 0; i < 64; ++i)
-                    if (!isxdigit(auth->password[i])) {
-                        g_set_error(error, NETPLAN_BACKEND_ERROR, NETPLAN_ERROR_UNSUPPORTED, "ERROR: %s: PSK length of 64 is only supported for hex-digit representation\n", id);
-                        return FALSE;
-                    }
-                /* this is required to be unquoted */
-                g_string_append_printf(s, "  psk=%s\n", auth->password);
-            } else if (len < 8 || len > 63) {
-                /* per wpa_supplicant spec, passphrase needs to be between 8
+
+    char* psk = NULL;
+    if (auth->psk)
+        psk = auth->psk;
+    else if (auth->password
+            && (auth->key_management == NETPLAN_AUTH_KEY_MANAGEMENT_WPA_PSK
+                || auth->key_management == NETPLAN_AUTH_KEY_MANAGEMENT_WPA_SAE)
+        )
+            psk = auth->password;
```

Comment:
praise: This is much more readable than the correpsonding `nm.c` code. Maybe it could be extracted into a helper function, to be used in both backends? (With a corresponding comment above that helper.)